app/services/builders/department_overview.rb

```diff
@@ -0,0 +1,57 @@
+module Builders
+  class DepartmentOverview < BaseService
+    include ModelsNamesHelper
+
+    def initialize(department)
+      @department = department
+    end
+
+    def call
+      department_overview = language_names.each_with_object({}) do |language, overview|
+        overview[language] = language_overview(language)
+      end
+
+      department_overview.with_indifferent_access
```

Comment:
Just sharing another shorter option...

```suggestion
      department_overview = language_names.index_with { |language|
        language_overview(language)
      }.with_indifferent_access
```

Reply:
True! I like it!